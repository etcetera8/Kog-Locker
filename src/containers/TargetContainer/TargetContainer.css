.new-segment-input {
  margin: 25px 5px 0 25px;
  padding: 5px;
  background: none;
  border: 1px solid #ff6400;
  color: #fff; }

.new-segment-input::placeholder {
  color: #aaa; }

.new-segment-button {
  font-size: 1.2em;
  cursor: pointer;
  background: none;
  border: none;
  color: #ff6400; }

.error-message {
  margin: 5px 0 0 25px;
  font-size: .8em;
  color: red; }
