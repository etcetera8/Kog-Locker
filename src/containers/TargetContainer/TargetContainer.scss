$orange: rgba(255,100,0,1);

.new-segment-input {
  margin: 25px 5px 0 25px;
  padding: 5px;
  border: 1px solid $orange;
  background: none;
  color: #fff;
}

.new-segment-input::placeholder {
  color: #aaa;
}

.new-segment-button {
  background: none;
  border: none;
  color: $orange;
  cursor: pointer;
  font-size: 1.2em;
}

.error-message {
  margin: 5px 0 0 25px;
  font-size: .8em;
  color: rgb(255, 0, 0);
}
