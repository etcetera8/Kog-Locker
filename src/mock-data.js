export const mockUser = {
  id: 9560317,
  username: null,
  resource_state: 3,
  firstname: "Parker",
  lastname: "Thomas",
  city: "Denver",
  state: "CO",
  country: "United States",
  sex: "M",
  premium: true,
  created_at: "2015-06-04T03:08:47Z",
  updated_at: "2018-02-11T15:51:13Z",
  badge_type_id: 1,
  profile_medium: "https://dgalywyr863hv.cloudfront.net/pictures/athletes/9560317/2891494/1/medium.jpg",
  profile: "https://dgalywyr863hv.cloudfront.net/pictures/athletes/9560317/2891494/1/large.jpg",
  friend: null,
  follower: null,
  email: "fugazi8@gmail.com",
  follower_count: 89,
  friend_count: 83,
  mutual_friend_count: 0,
  athlete_type: 0,
  date_preference: "%m/%d/%Y",
  measurement_preference: "feet",
  clubs: [
  {
  id: 121895,
  resource_state: 2,
  name: "Team Colavita Parisi Coffee",
  profile_medium: "https://dgalywyr863hv.cloudfront.net/pictures/clubs/121895/2306445/1/medium.jpg",
  profile: "https://dgalywyr863hv.cloudfront.net/pictures/clubs/121895/2306445/1/large.jpg",
  cover_photo: "https://dgalywyr863hv.cloudfront.net/pictures/clubs/121895/5683064/1/large.jpg",
  cover_photo_small: "https://dgalywyr863hv.cloudfront.net/pictures/clubs/121895/5683064/1/small.jpg",
  sport_type: "cycling",
  city: "Kansas City",
  state: "Missouri",
  country: "United States",
  private: true,
  member_count: 42,
  featured: false,
  verified: false,
  url: "tcpc-kc",
  membership: "member",
  admin: false,
  owner: false
  }
  ],
  ftp: 281,
  weight: 69.8532,
  bikes: [
  {
  id: "b4506235",
  primary: false,
  name: "Race",
  resource_state: 2,
  distance: 0
  },
  {
  id: "b4506234",
  primary: true,
  name: "TallBoy",
  resource_state: 2,
  distance: 0
  }
  ],
  shoes: [ ]
}

export const mockSegment = {
  id: 609371,
  resource_state: 3,
  name: "Lookout Mountain (Gate-BB Grave)",
  activity_type: "Ride",
  distance: 7361.19,
  average_grade: 5.4,
  maximum_grade: 31.1,
  elevation_high: 2230.3,
  elevation_low: 1843,
  start_latlng: [
  39.740329,
  -105.227683
  ],
  end_latlng: [
  39.731674,
  -105.239256
  ],
  start_latitude: 39.740329,
  start_longitude: -105.227683,
  end_latitude: 39.731674,
  end_longitude: -105.239256,
  climb_category: 3,
  city: "Golden",
  state: "Colorado",
  country: "United States",
  private: false,
  hazardous: false,
  starred: false,
  created_at: "2010-04-16T03:33:52Z",
  updated_at: "2018-02-12T09:11:49Z",
  total_elevation_gain: 0,
  map: {
  id: "s609371",
  polyline: "_xpqF`hgaS\QHDFJCNSb@Ul@Q`AY|BA\Dl@Rt@PdA@f@E\IRKJuAr@UXSr@SpAYr@U\o@r@yAdAw@p@gAlAS^Kb@An@Jx@Nz@?h@SVIBa@EcBy@g@Oq@Ac@Je@^S`@G^?v@Bj@R`B@ZKZKB[C{@g@a@Oa@I]A]Bk@R_@\{@nAYXSNa@LaACWHGDU\Mj@EvAKd@[`@UDcC@_@Dk@PUNOPIN]hAm@xAEN@~@NxAHb@P^x@hAXv@XdBNnAB|@Eh@]bBI~@CvAO|B?HJ^H@Z[p@iEb@gBFg@D}ABERAHV@n@Ct@GvAOnB@LPVJ?RODYHwAh@wCDq@@g@C_AQu@eAcBQi@Gg@@oAFe@HUVWXGn@Bl@@XIbAo@VIZAn@Db@EZQnAiA\SRGZC`@@ZFzAXh@PPNZd@^^\NnA^XFZ@\OJKz@_B`@WPAJ@NDx@t@|Af@`@`@Vp@FTnA|GRp@Xh@pAlAX`@ZZRFV@^El@QTCJ@\JdAz@`@`@V^x@tA\r@xApDX^LHLBJ?VGPM`AeAf@e@p@e@d@Sd@?pANfDVpAArAI\?JL@LEHKJYDm@Ec@BM`@JPVHr@A`@BDF@PEJSJq@Ae@I]CwBFoBPWFW`@?LHPVLT?fAM`@@d@LzAz@VFZBn@Cr@Un@_@hBoBZ_@Rk@Jo@L{BLs@HWLSn@e@LMHg@Si@u@s@i@]uAo@]WK]IiBWgBDe@Z{@?g@GKSQm@S_Ag@cBc@U[C_@H{AHe@N]jCoDh@m@POVKTGt@Ah@Br@JPFHHXf@DRBPAr@ShCCp@@h@",
  resource_state: 3
  },
  effort_count: 130710,
  athlete_count: 14830,
  star_count: 1031,
  athlete_segment_stats: {
    pr_elapsed_time: 1383,
    pr_date: "2017-09-17",
    effort_count: 8
  }
}

export const mockUserStats = {
  biggest_ride_distance: 174857,
  biggest_climb_elevation_gain: 1271,
  recent_ride_totals: {
  count: 3,
  distance: 91564.900390625,
  moving_time: 18456,
  elapsed_time: 23560,
  elevation_gain: 2166,
  achievement_count: 58
  },
  recent_run_totals: {
  count: 2,
  distance: 7873.5,
  moving_time: 3106,
  elapsed_time: 3516,
  elevation_gain: 69.58277893066406,
  achievement_count: 11
  },
  recent_swim_totals: {
  count: 0,
  distance: 0,
  moving_time: 0,
  elapsed_time: 0,
  elevation_gain: 0,
  achievement_count: 0
  },
  ytd_ride_totals: {
  count: 7,
  distance: 240759,
  moving_time: 46670,
  elapsed_time: 55484,
  elevation_gain: 4847
  },
  ytd_run_totals: {
  count: 3,
  distance: 10560,
  moving_time: 4048,
  elapsed_time: 4533,
  elevation_gain: 97
  },
  ytd_swim_totals: {
  count: 0,
  distance: 0,
  moving_time: 0,
  elapsed_time: 0,
  elevation_gain: 0
  },
  all_ride_totals: {
  count: 612,
  distance: 20215516,
  moving_time: 3290788,
  elapsed_time: 3987884,
  elevation_gain: 250799
  },
  all_run_totals: {
  count: 13,
  distance: 56967,
  moving_time: 20243,
  elapsed_time: 23761,
  elevation_gain: 203
  },
  all_swim_totals: {
  count: 4,
  distance: 2409,
  moving_time: 2439,
  elapsed_time: 2444,
  elevation_gain: 0
  }
}