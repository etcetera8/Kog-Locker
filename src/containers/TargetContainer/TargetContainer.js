import React, { Component } from 'react';
import { connect } from 'react-redux';
import MapCard from '../../components/MapCard/MapCard';
import { GoogleApiWrapper } from 'google-maps-react';
import {gKey} from '../../apiKey.js'
import { addUserTarget } from '../../actions/actionIndex'
import { segmentCall } from '../../api';
import './TargetContainer.css';

class TargetContainer extends Component {
  constructor(props) {
    super(props)
    this.state = {
      segmentId: '',
      segmentError: false,
      loading: false
    }
  }

  changeHelper = (e) => {
    this.setState({segmentId: e.target.value })
  }

  helper = async (e) => {
    //15990214
    e.preventDefault();
    const response = await segmentCall(this.state.segmentId)
    console.log(response);
    if (response.message) {
      this.setState({ segmentError: true, segmentId: '' })
      console.log("im a type error")
    } else {
      this.setState({ segmentError: false, segmentId: '' })
      this.props.setUserTarget(response);
      localStorage.setItem('target', JSON.stringify(response));
    }
  }
  
  render() {
    const {average_grade, distance, elevation_high, elevation_low, athlete_segment_stats} = this.props.userTarget
    
    return (
      <main>
        <MapCard 
          google={this.props.google}
          userTarget={this.props.userTarget}
        />

        <div className='card'>
          <span className='card-data'>Grade: <span className='nums'>{average_grade}%</span></span>
          <span className='card-data'>Distance: <span className='nums'>{distance} miles</span></span>
          <span className='card-data'>Elevation Gain: <span className='nums'>{(elevation_high - elevation_low)} ft</span></span>
          {athlete_segment_stats && 
          <span className='card-data'>Number of attempts: <span className='nums'>{athlete_segment_stats.effort_count} </span></span>
          }

          <form>
            <input
              className="new-segment-input"
              type='text' 
              placeholder="Add new segment ID#" 
              value={this.state.segmentId}
              onChange={this.changeHelper}
            />

            <button 
              onClick={this.helper}
              className="new-segment-button"
              >
                <i class="fas fa-plus"></i>
            </button>
            {this.state.segmentError && 
              <p className="error-message">You must put in a valid segment ID</p>
            }
          </form>
        </div>
      </main>
    )
  }
  
}

const mapStateToProps = (state) => ({
  userTarget: state.userTarget
})

const mapDispatchToProps = (dispatch) => ({
  setUserTarget: (data) => dispatch(addUserTarget(data))
})

const wrapper = GoogleApiWrapper({apiKey:gKey})(TargetContainer)

export default connect(mapStateToProps, mapDispatchToProps)(wrapper)