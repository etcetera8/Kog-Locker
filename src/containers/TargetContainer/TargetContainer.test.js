import React from 'react';
import {shallow} from 'enzyme';
import { TargetContainer, mapStateToProps, mapDispatchToProps } from './TargetContainer';
import { cleanSegment } from '../../mock-data.js'

describe("targetContainer", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<TargetContainer userTarget={cleanSegment}
                                       google={{maps: "Google Maps API"}}/>)
  })

  it('should start out with a default state', () => {
    const state = {input: '', segmentError: false, loading: false}
    expect(wrapper.state()).toEqual(state)
  })

  it('the changeHelper function should set the state', () => {
    const mockEvent = {target: {value:'123'}}
    expect(wrapper.state().input).toEqual('')
    wrapper.find('input').first().simulate('change', mockEvent)
    expect(wrapper.state().input).toEqual('123')
  })

  it("mapDispatchToProps should return an object with the right keys", () => {
    const results = mapDispatchToProps();
    expect(Object.keys(results)).toEqual(["setUserTarget"])
  })

  it("mapStateToProps should return an object with correct keys", () => {
    const mockState = {userTarget: "Target segment"};
    const results = mapStateToProps(mockState);
    expect(results).toEqual({userTarget: "Target segment"})
  })
})