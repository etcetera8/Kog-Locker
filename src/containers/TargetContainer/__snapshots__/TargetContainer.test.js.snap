// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`targetContainer should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TargetContainer
    google={
        Object {
            "maps": "Google Maps API",
          }
    }
    setUserTarget={[Function]}
    userTarget={
        Object {
            "athlete_segment_stats": Object {
              "effort_count": 8,
              "goalTime": 22.3585,
              "pr_date": "2017-09-17",
              "pr_elapsed_time": 1383,
            },
            "average_grade": 5.4,
            "city": "Golden",
            "distance": 4,
            "elevation_high": 7155,
            "elevation_low": 5912,
            "end_latlng": Array [
              39.731674,
              -105.239256,
            ],
            "name": "Lookout Mountain (Gate-BB Grave)",
            "polyline": "_xpqF",
            "total_elevation_gain": 0,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(MapCard)
          google={
                    Object {
                              "maps": "Google Maps API",
                            }
          }
          userTarget={
                    Object {
                              "athlete_segment_stats": Object {
                                "effort_count": 8,
                                "goalTime": 22.3585,
                                "pr_date": "2017-09-17",
                                "pr_elapsed_time": 1383,
                              },
                              "average_grade": 5.4,
                              "city": "Golden",
                              "distance": 4,
                              "elevation_high": 7155,
                              "elevation_low": 5912,
                              "end_latlng": Array [
                                39.731674,
                                -105.239256,
                              ],
                              "name": "Lookout Mountain (Gate-BB Grave)",
                              "polyline": "_xpqF",
                              "total_elevation_gain": 0,
                            }
          }
/>,
        <div
          className="card"
>
          <span
                    className="card-data"
          >
                    Grade: 
                    <span
                              className="nums"
                    >
                              5.4
                              %
                    </span>
          </span>
          <span
                    className="card-data"
          >
                    Distance: 
                    <span
                              className="nums"
                    >
                              4
                               miles
                    </span>
          </span>
          <span
                    className="card-data"
          >
                    Elevation Gain: 
                    <span
                              className="nums"
                    >
                              1243
                               ft
                    </span>
          </span>
          <span
                    className="card-data"
          >
                    Number of attempts: 
                    <span
                              className="nums"
                    >
                              8
                               
                    </span>
          </span>
          <form>
                    <input
                              className="new-segment-input"
                              onChange={[Function]}
                              placeholder="Add new segment ID#"
                              type="text"
                              value=""
                    />
                    <button
                              className="new-segment-button"
                              onClick={[Function]}
                    >
                              <i
                                        className="fas fa-plus"
                              />
                    </button>
          </form>
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "google": Object {
            "maps": "Google Maps API",
          },
          "userTarget": Object {
            "athlete_segment_stats": Object {
              "effort_count": 8,
              "goalTime": 22.3585,
              "pr_date": "2017-09-17",
              "pr_elapsed_time": 1383,
            },
            "average_grade": 5.4,
            "city": "Golden",
            "distance": 4,
            "elevation_high": 7155,
            "elevation_low": 5912,
            "end_latlng": Array [
              39.731674,
              -105.239256,
            ],
            "name": "Lookout Mountain (Gate-BB Grave)",
            "polyline": "_xpqF",
            "total_elevation_gain": 0,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="card-data"
>
              Grade: 
              <span
                            className="nums"
              >
                            5.4
                            %
              </span>
</span>,
            <span
              className="card-data"
>
              Distance: 
              <span
                            className="nums"
              >
                            4
                             miles
              </span>
</span>,
            <span
              className="card-data"
>
              Elevation Gain: 
              <span
                            className="nums"
              >
                            1243
                             ft
              </span>
</span>,
            <span
              className="card-data"
>
              Number of attempts: 
              <span
                            className="nums"
              >
                            8
                             
              </span>
</span>,
            <form>
              <input
                            className="new-segment-input"
                            onChange={[Function]}
                            placeholder="Add new segment ID#"
                            type="text"
                            value=""
              />
              <button
                            className="new-segment-button"
                            onClick={[Function]}
              >
                            <i
                                          className="fas fa-plus"
                            />
              </button>
</form>,
          ],
          "className": "card",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Grade: ",
                <span
                  className="nums"
>
                  5.4
                  %
</span>,
              ],
              "className": "card-data",
            },
            "ref": null,
            "rendered": Array [
              "Grade: ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    5.4,
                    "%",
                  ],
                  "className": "nums",
                },
                "ref": null,
                "rendered": Array [
                  5.4,
                  "%",
                ],
                "type": "span",
              },
            ],
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Distance: ",
                <span
                  className="nums"
>
                  4
                   miles
</span>,
              ],
              "className": "card-data",
            },
            "ref": null,
            "rendered": Array [
              "Distance: ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    4,
                    " miles",
                  ],
                  "className": "nums",
                },
                "ref": null,
                "rendered": Array [
                  4,
                  " miles",
                ],
                "type": "span",
              },
            ],
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Elevation Gain: ",
                <span
                  className="nums"
>
                  1243
                   ft
</span>,
              ],
              "className": "card-data",
            },
            "ref": null,
            "rendered": Array [
              "Elevation Gain: ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    1243,
                    " ft",
                  ],
                  "className": "nums",
                },
                "ref": null,
                "rendered": Array [
                  1243,
                  " ft",
                ],
                "type": "span",
              },
            ],
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Number of attempts: ",
                <span
                  className="nums"
>
                  8
                   
</span>,
              ],
              "className": "card-data",
            },
            "ref": null,
            "rendered": Array [
              "Number of attempts: ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    8,
                    " ",
                  ],
                  "className": "nums",
                },
                "ref": null,
                "rendered": Array [
                  8,
                  " ",
                ],
                "type": "span",
              },
            ],
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  className="new-segment-input"
                  onChange={[Function]}
                  placeholder="Add new segment ID#"
                  type="text"
                  value=""
/>,
                <button
                  className="new-segment-button"
                  onClick={[Function]}
>
                  <i
                                    className="fas fa-plus"
                  />
</button>,
                false,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "new-segment-input",
                  "onChange": [Function],
                  "placeholder": "Add new segment ID#",
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <i
                    className="fas fa-plus"
/>,
                  "className": "new-segment-button",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fas fa-plus",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                "type": "button",
              },
              false,
            ],
            "type": "form",
          },
        ],
        "type": "div",
      },
    ],
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(MapCard)
            google={
                        Object {
                                    "maps": "Google Maps API",
                                  }
            }
            userTarget={
                        Object {
                                    "athlete_segment_stats": Object {
                                      "effort_count": 8,
                                      "goalTime": 22.3585,
                                      "pr_date": "2017-09-17",
                                      "pr_elapsed_time": 1383,
                                    },
                                    "average_grade": 5.4,
                                    "city": "Golden",
                                    "distance": 4,
                                    "elevation_high": 7155,
                                    "elevation_low": 5912,
                                    "end_latlng": Array [
                                      39.731674,
                                      -105.239256,
                                    ],
                                    "name": "Lookout Mountain (Gate-BB Grave)",
                                    "polyline": "_xpqF",
                                    "total_elevation_gain": 0,
                                  }
            }
/>,
          <div
            className="card"
>
            <span
                        className="card-data"
            >
                        Grade: 
                        <span
                                    className="nums"
                        >
                                    5.4
                                    %
                        </span>
            </span>
            <span
                        className="card-data"
            >
                        Distance: 
                        <span
                                    className="nums"
                        >
                                    4
                                     miles
                        </span>
            </span>
            <span
                        className="card-data"
            >
                        Elevation Gain: 
                        <span
                                    className="nums"
                        >
                                    1243
                                     ft
                        </span>
            </span>
            <span
                        className="card-data"
            >
                        Number of attempts: 
                        <span
                                    className="nums"
                        >
                                    8
                                     
                        </span>
            </span>
            <form>
                        <input
                                    className="new-segment-input"
                                    onChange={[Function]}
                                    placeholder="Add new segment ID#"
                                    type="text"
                                    value=""
                        />
                        <button
                                    className="new-segment-button"
                                    onClick={[Function]}
                        >
                                    <i
                                                className="fas fa-plus"
                                    />
                        </button>
            </form>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "google": Object {
              "maps": "Google Maps API",
            },
            "userTarget": Object {
              "athlete_segment_stats": Object {
                "effort_count": 8,
                "goalTime": 22.3585,
                "pr_date": "2017-09-17",
                "pr_elapsed_time": 1383,
              },
              "average_grade": 5.4,
              "city": "Golden",
              "distance": 4,
              "elevation_high": 7155,
              "elevation_low": 5912,
              "end_latlng": Array [
                39.731674,
                -105.239256,
              ],
              "name": "Lookout Mountain (Gate-BB Grave)",
              "polyline": "_xpqF",
              "total_elevation_gain": 0,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="card-data"
>
                Grade: 
                <span
                                className="nums"
                >
                                5.4
                                %
                </span>
</span>,
              <span
                className="card-data"
>
                Distance: 
                <span
                                className="nums"
                >
                                4
                                 miles
                </span>
</span>,
              <span
                className="card-data"
>
                Elevation Gain: 
                <span
                                className="nums"
                >
                                1243
                                 ft
                </span>
</span>,
              <span
                className="card-data"
>
                Number of attempts: 
                <span
                                className="nums"
                >
                                8
                                 
                </span>
</span>,
              <form>
                <input
                                className="new-segment-input"
                                onChange={[Function]}
                                placeholder="Add new segment ID#"
                                type="text"
                                value=""
                />
                <button
                                className="new-segment-button"
                                onClick={[Function]}
                >
                                <i
                                                className="fas fa-plus"
                                />
                </button>
</form>,
            ],
            "className": "card",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Grade: ",
                  <span
                    className="nums"
>
                    5.4
                    %
</span>,
                ],
                "className": "card-data",
              },
              "ref": null,
              "rendered": Array [
                "Grade: ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      5.4,
                      "%",
                    ],
                    "className": "nums",
                  },
                  "ref": null,
                  "rendered": Array [
                    5.4,
                    "%",
                  ],
                  "type": "span",
                },
              ],
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Distance: ",
                  <span
                    className="nums"
>
                    4
                     miles
</span>,
                ],
                "className": "card-data",
              },
              "ref": null,
              "rendered": Array [
                "Distance: ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      4,
                      " miles",
                    ],
                    "className": "nums",
                  },
                  "ref": null,
                  "rendered": Array [
                    4,
                    " miles",
                  ],
                  "type": "span",
                },
              ],
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Elevation Gain: ",
                  <span
                    className="nums"
>
                    1243
                     ft
</span>,
                ],
                "className": "card-data",
              },
              "ref": null,
              "rendered": Array [
                "Elevation Gain: ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      1243,
                      " ft",
                    ],
                    "className": "nums",
                  },
                  "ref": null,
                  "rendered": Array [
                    1243,
                    " ft",
                  ],
                  "type": "span",
                },
              ],
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Number of attempts: ",
                  <span
                    className="nums"
>
                    8
                     
</span>,
                ],
                "className": "card-data",
              },
              "ref": null,
              "rendered": Array [
                "Number of attempts: ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      8,
                      " ",
                    ],
                    "className": "nums",
                  },
                  "ref": null,
                  "rendered": Array [
                    8,
                    " ",
                  ],
                  "type": "span",
                },
              ],
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <input
                    className="new-segment-input"
                    onChange={[Function]}
                    placeholder="Add new segment ID#"
                    type="text"
                    value=""
/>,
                  <button
                    className="new-segment-button"
                    onClick={[Function]}
>
                    <i
                                        className="fas fa-plus"
                    />
</button>,
                  false,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "new-segment-input",
                    "onChange": [Function],
                    "placeholder": "Add new segment ID#",
                    "type": "text",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <i
                      className="fas fa-plus"
/>,
                    "className": "new-segment-button",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "fas fa-plus",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                  "type": "button",
                },
                false,
              ],
              "type": "form",
            },
          ],
          "type": "div",
        },
      ],
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
